longest-run.playground/Contents.swift

```diff
@@ -7,23 +7,68 @@ This excercise lends itself to TDD and we have provided a few basic tests below.
 import Cocoa
 
 func longestRun(whole: String) -> [String] {
-    return [whole]
+    
+    let allRuns = stringToRunsList(whole)
+    let longestRuns = findLongestRuns(allRuns)
+    let longestRunStrings = longestRuns.map({String($0)})
+    return longestRunStrings
+}
+
+func stringToRunsList(whole: String) -> [[Character]] {
+    
+    var chars = [[Character]]()
+    
+    for index in whole.characters.indices {
+
+        let daChar = whole[index]
+        let onDaRun = chars.last?.contains(daChar)
+
+        // nil coalescing operator for da win
+        if (onDaRun ?? false)  {
+            chars[chars.endIndex.predecessor()].append(daChar)
+        } else {
+            chars.append([daChar])
+        }
+    }
+    return chars
+}
+
+func findLongestRuns(runs: [[Character]]) -> [[Character]] {
+    
+    let longestRunCount = runs.reduce(0, combine: { $0 > $1.count ? $0 : $1.count })
```

Comment:
Definitely very cool. I was pondering whether it's possible to use reduce to solve this problem in "one line" but got stuck